### PR TITLE
CORE-322 replicate reply hash check

### PR DESCRIPTION
**Pull request by esclapes**
This PR adds a new function `compareReplyHash` which replicates the compare logic [as used by reply](https://github.com/bufferapp/respond-server/blob/118e04ad18b1cf5f55bee7fa828a3e580608c640/server/services/authentication-service.js#L147). 

* **Note that I am not using the same dependencies as reply, I am using similar packages that were already in the service to avoid adding unecessary dependencies**
* Unit tests are based on real hashes from `reply-dev` database, jut to be sure
* Updated jest to use `test.each` :smile: 

~This PR can be merged to master as the new function is not being used and the only other change is extracting `compare` and `hash` functions to its own file `./lib/bcryptAsync.js`~ `compare` and `hash` support promises already.

### CODE REVIEW

**Comment by esclapes** [comment]
According to `bcryptjs` [docs](https://github.com/dcodeIO/bcrypt.js#hashs-salt-callback-progresscallback), both `hash` and `compare` return a Promise if no callback is passed (as of `2.4.0`), as we do use latest `2.4.3`, we can drop the use of `promisify` and use the functions directly. Commit coming in a second.

**Comment by esclapes** [comment]
@kiriappeee I think you mentioned that the check for `bufferbotbrains/cicd-authentication-service ` should not be getting stuck... Would you mind checking it whenever you have some bandwidth?

File: src/libs/compareReplyHash.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const crypto = require('crypto')
```
Comment (djfarrelly) [inline src/libs/compareReplyHash.js]: 👍 On using the standard crypto package

File: src/models/authenticationAccountModel.js [hunk 2]
```diff
@@ -4,15 +4,11 @@ const {
...
 } = require('../libs/errorCodes')
 const uuid = require('uuid/v4')
-const bcrypt = require('bcryptjs')
-const { promisify } = require('util')
 const mongoose = require('mongoose')
 require('mongoose-type-email')
 
 const { Schema, SchemaTypes } = mongoose
-const hash = promisify(bcrypt.hash)
-const compare = promisify(bcrypt.compare)
```
Comment (djfarrelly) [inline src/models/authenticationAccountModel.js]: Nice change being able to drop `promisify` here, definitely cleaner!

**Pull request merged**